interfaces/repo_test.go

```diff
@@ -43,26 +44,71 @@ var _ = Suite(&RepositorySuite{
 })
 
 func (s *RepositorySuite) SetUpTest(c *C) {
-	s.plug = &Plug{
-		Snap:      "provider",
-		Name:      "plug",
-		Interface: "interface",
-		Label:     "label",
-		Attrs:     map[string]interface{}{"attr": "value"},
-		Apps:      []string{"meta/hooks/plug"},
-	}
-	s.slot = &Slot{
-		Snap:      "consumer",
-		Name:      "slot",
-		Interface: "interface",
-		Label:     "label",
-		Attrs:     map[string]interface{}{"attr": "value"},
-		Apps:      []string{"app"},
-	}
+	// NOTE: the names provider/consumer are confusing. They will be fixed shortly.
+	provider, err := snap.InfoFromSnapYaml([]byte(`
+name: provider
+apps:
+    meta/hooks/plug:
+plugs:
+    plug:
+        interface: interface
+        label: label
+        attr: value
+`))
+	c.Assert(err, IsNil)
+	s.plug = &Plug{PlugInfo: provider.Plugs["plug"]}
+	consumer, err := snap.InfoFromSnapYaml([]byte(`
+name: consumer
+apps:
+    app:
+slots:
+    slot:
+        interface: interface
+        label: label
+        attr: value
+`))
+	c.Assert(err, IsNil)
+	s.slot = &Slot{SlotInfo: consumer.Slots["slot"]}
 	s.emptyRepo = NewRepository()
 	s.testRepo = NewRepository()
-	err := s.testRepo.AddInterface(s.iface)
+	err = s.testRepo.AddInterface(s.iface)
+	c.Assert(err, IsNil)
+}
+
+// plugFromYaml parses the given Yaml and returns Plug with the given name.
+func plugFromYaml(c *C, plugName string, yaml []byte) *Plug {
+	info, err := snap.InfoFromSnapYaml(yaml)
 	c.Assert(err, IsNil)
+	plugInfo := info.Plugs[plugName]
+	c.Assert(plugInfo, Not(IsNil))
+	return &Plug{PlugInfo: plugInfo}
+}
+
+// slotFromYaml parses the given Yaml and returns Slot with the given name.
+func slotFromYaml(c *C, slotName string, yaml []byte) *Slot {
+	info, err := snap.InfoFromSnapYaml(yaml)
+	c.Assert(err, IsNil)
+	slotInfo := info.Slots[slotName]
+	c.Assert(slotInfo, Not(IsNil))
+	return &Slot{SlotInfo: slotInfo}
+}
+
+func addPlugsAndSlotsFromYaml(c *C, repo *Repository, yamls ...[]byte) []*snap.Info {
```

Comment:
Can we please change this signature to:

`func addPlugsSlots(c *C, repo *Repository, yamls ...string)`

Having strings will prevent you from having to convert everywhere else. And the short name is much more readable than the longer one. It's obvious from context that it is yaml.